File: web/src/api/basicDict.ts
import { http,resData } from "@/utils/http";
 
// @Tags BasicDict
// @Summary 创建BasicDict
export const createBasicDict = (data?: object) => {
   return http.request<resData>("post", "/basicDict/createBasicDict", { data });  
}

// @Tags BasicDict
// @Summary 删除BasicDict
export const deleteBasicDict = (data?: object) => {
   return http.request<resData>("post", "/basicDict/deleteBasicDict", { data });  
}
 

// @Tags BasicDict
// @Summary 删除ByIdsBasicDict
export const deleteBasicDictByIds = (data?: object) => {
   return http.request<resData>("post", "/basicDict/deleteBasicDictByIds", { data });  
}
 

// @Tags BasicDict
// @Summary 更新BasicDict
export const updateBasicDict = (data?: object) => {
   return http.request<resData>("post", "/basicDict/updateBasicDict", { data });  
}

// @Tags BasicDict
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
   return http.request<resData>("post", "/basicDict/quickEdit", { data });  
}
 

// @Tags BasicDict
// @Summary 用id查询BasicDict
export const findBasicDict = (params?: object) => {
   return http.request<resData>("get", "/basicDict/findBasicDict", { params });  
}
 

// @Tags BasicDict
// @Summary 分页获取BasicDict列表
export const getBasicDictList = (params?: object) => {
   return http.request<resData>("get", "/basicDict/getBasicDictList", { params });  
}
 

// @Tags BasicDict
// @Summary 导出excelBasicDict列表
export const excelList = (params?: object) => {
   return http.request<resData>("get", "/basicDict/excelList", { params });  
}

File: server/router/business/basicDict.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type BasicDictRouter struct {
}

// InitBasicDictRouter 初始化 BasicDict 路由信息
func (s *BasicDictRouter) InitBasicDictRouter(Router *gin.RouterGroup) {
	router := Router.Group("basicDict").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("basicDict")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.BasicDictApi
	{
		router.POST("createBasicDict", apiV1.CreateBasicDict)   // 新建BasicDict
		router.POST("deleteBasicDict", apiV1.DeleteBasicDict) // 删除BasicDict
		router.POST("deleteBasicDictByIds", apiV1.DeleteBasicDictByIds) // 批量删除BasicDict
		router.POST("updateBasicDict", apiV1.UpdateBasicDict)    // 更新BasicDict
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findBasicDict", apiV1.FindBasicDict)        // 根据ID获取BasicDict
		routerNoRecord.GET("getBasicDictList", apiV1.GetBasicDictList)  // 获取BasicDict列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel BasicDict列表
	}
}
